cmd/controller-manager/app/controller-manager.go

```diff
@@ -100,6 +100,17 @@ func Run(opts *options.Options) error {
 
 	stopChan := setupSignalHandler()
 
+	// TODO(sohankunkerkar) Remove when no longer necessary.
```

Comment:
Please move lines 103-113 to the conditional block starting on line 261.  It should only be executed if the scope in the federation config is empty rather than on every invocation of the controller manager.